packages/hurumap-core/src/ShareButton/ShareButton.js

```diff
@@ -0,0 +1,58 @@
+import { Box, SvgIcon } from "@mui/material";
+import React from "react";
+import {
+  TwitterShareButton,
+  LinkedinShareButton,
+  FacebookShareButton,
+  EmailShareButton,
+  WhatsappShareButton,
+  TelegramShareButton,
+  PinterestShareButton,
+} from "react-share";
+
+const componentMap = {
+  Facebook: FacebookShareButton,
+  Twitter: TwitterShareButton,
+  LinkedIn: LinkedinShareButton,
+  WhatsApp: WhatsappShareButton,
+  Email: EmailShareButton,
+  Telegram: TelegramShareButton,
+  Pinterest: PinterestShareButton,
+};
+
+const ShareButton = React.forwardRef(function ShareButton({
+  name,
+  url,
+  ButtonProps,
+  icon,
+  ...props
```

Comment:
```suggestion
  ButtonProps,
  IconPros,
  icon,
  name,
  url,
  ...props
```

1. `IconProps` allow customisation of `SvgIcon`.
2. Always _sort_ props.